scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1330,27 +1330,18 @@ static void task_uncharge_qrt(struct task_ctx *taskc)
 	taskc->qrt_llc_id = MAX_LLCS;
 }
 
-static void kick_idle_cpu(struct task_struct *p, struct layer *layer)
+static void layer_kick_idle_cpu(struct layer *layer)
 {
-	const struct cpumask *idle_smtmask;
+	const struct cpumask *layer_cpumask, *idle_smtmask;;
 	s32 cpu;
 
-	if (!(idle_smtmask = scx_bpf_get_idle_smtmask()))
+	if (!(layer_cpumask = lookup_layer_cpumask(layer->id)) ||
+	    !(idle_smtmask = scx_bpf_get_idle_smtmask()))
```

Comment:
I wonder if `scx_bpf_get_idle_cpumask` would help, especially when a machine is mostly saturated. So maybe have a param to control whether idle smt or idle cpu is used. Or try some clever heuristic based on load.